### PR TITLE
e2e: add test covering cronjob-scheduled-timestamp annotation added by cronjob

### PR DESCRIPTION
**Pull request by soltysh**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This is part of https://github.com/kubernetes/enhancements/issues/4026

#### Which issue(s) this PR fixes:
None

#### Special notes for your reviewer:
/assign @atiratree

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/issues/4026
```

### CODE REVIEW

**Review by mimowo — COMMENTED** [review]
LGTM

File: test/e2e/apps/cronjob.go [hunk 1]
```diff
@@ -261,6 +261,26 @@ var _ = SIGDescribe("CronJob", func() {
...
+			nil, nil, nil)
+		cronJob, err := createCronJob(ctx, f.ClientSet, f.Namespace.Name, cronJob)
+		framework.ExpectNoError(err, "Failed to create CronJob in namespace %s", f.Namespace.Name)
+
+		ginkgo.By("Ensuring CronJobsScheduledAnnotation annotation exists on the newely created job")
+		err = waitForJobsAtLeast(ctx, f.ClientSet, f.Namespace.Name, 1)
+		framework.ExpectNoError(err, "Failed to ensure a job exists in namespace %s", f.Namespace.Name)
+
+		jobs, err := f.ClientSet.BatchV1().Jobs(f.Namespace.Name).List(ctx, metav1.ListOptions{})
+		gomega.Expect(jobs.Items[0].Annotations).Should(gomega.HaveKey(batchv1.CronJobScheduledTimestampAnnotation),
```
Comment (atiratree) [inline test/e2e/apps/cronjob.go:276]: I know we have a unit test for this, but we could at least test that it is possible to parse the timestamp

**Pull request merged**